aleph-client/src/connections.rs

```diff
@@ -12,21 +12,72 @@ use subxt::{
     SubstrateConfig,
 };
 
-use crate::{api, sp_weights::weight_v2::Weight, BlockHash, Call, Client, KeyPair, TxStatus};
+use crate::{
+    api, sp_weights::weight_v2::Weight, AccountId, BlockHash, Call, KeyPair, SubxtClient, TxStatus,
+};
 
 #[derive(Clone)]
 pub struct Connection {
-    pub client: Client,
+    client: SubxtClient,
 }
 
 pub struct SignedConnection {
-    pub connection: Connection,
-    pub signer: KeyPair,
+    connection: Connection,
+    signer: KeyPair,
 }
 
+#[derive(Clone)]
 pub struct RootConnection {
-    pub connection: Connection,
-    pub root: KeyPair,
+    connection: SignedConnection,
+}
+
+pub trait AsConnection: Sync {
```

Comment:
I think this trait can be just internal to this crate? `Connection` except new, has no public API. You will need to remove it from other traits as well. Actually what you need here is something like `AsSubxtClient`?

Reply:
+ I'm not sure why we should hide potentially useful trait